runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -500,8 +503,10 @@ static StreamingDataflowWorker forTesting(
       boolean publishCounters,
       HotKeyLogger hotKeyLogger,
       Supplier<Instant> clock,
-      Function<String, ScheduledExecutorService> executorSupplier) {
-    BoundedQueueExecutor boundedQueueExecutor = createWorkUnitExecutor(options);
+      Function<String, ScheduledExecutorService> executorSupplier,
+      BoundedQueueExecutor executor) {
```

Comment:
how about naming workUnitExecutor and can just use same variable below to avoid proliferation of names

if (workUnitExeuctor == null) workUnitExecutor = createWorkUnitExecutor(options);